integrate/romberg.go

```diff
@@ -0,0 +1,98 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import (
+	"math"
+)
+
+//
+// Helper method for getting the base 2 exponent of an integer
+//
+func getBase2Exponent(x int) int {
+	n := 1
+	k := 0
+
+	for n < x {
+		n <<= 1
+		k += 1
+	}
+	return k
+}
+
+// Romberg estimates the integral of the function f
+// \int_0^1 f(x)dx
+// The function f must be sampled at equal intervals.
+// Romberg's method approximates the integral of the function by
+// applying Richardson extrapolation. Hence, the function adaptively
+// increase the number of evaluation points until the desired accuracy
+// is obtained or the maximum number of evaluation points has been used.
+// See https://en.wikipedia.org/wiki/Romberg%27s_method for a thorough description
+// of the algorithm.
+//
+// The length of the length of the f array must be 2**k + 1, where k is a positive integer
+// The minimum length of the work array is 2*k
+// The method returns the value of the integral, and the number of levels needed to reach
+// the desired accuracy
+func Romberg(f []float64, acc float64, work []float64) (float64, int) {
+	ninterv := len(f) - 1
+	// Check that number of intervals is equal to a power of 2
+	dx := 1.0
+
+	k := getBase2Exponent(ninterv)
+
+	n := 1 << uint(k)
+	if n != ninterv {
+		panic("The length of the function array passed has to be 1 + 2^k. For some integer k.")
+	}
+
+	if len(work) < 2*k {
+		panic("The length of the work array has to be at least 2*k, when len(f) = 2^k + 1")
+	}
+
+	Rprev := work[:k]
+	Rcur := work[k:]
+
+	Rprev[0] = (f[0] + f[len(f)-1]) * 0.5 * dx
+
+	for i := 1; i < k; i++ {
+		dx /= 2.0
+		estimate := 0.0
+		step := ninterv >> uint32(i)
+
+		for j := 0; j < ninterv/2; j += step {
+			estimate += f[2*j+step]
+		}
+
+		// Calculate first element on new row
+		Rcur[0] = estimate*dx + 0.5*Rprev[0]
+
+		// Calculate the rest of the entries
+		for j := 1; j <= i; j++ {
+			factor := math.Pow(4, float64(j))
+			Rcur[j] = (factor*Rcur[j-1] - Rprev[j-1]) / (factor - 1)
+		}
+
+		if math.Abs(Rprev[i-1]-Rcur[i]) < acc {
+			return Rcur[i], i
+		}
+
+		// Swap buffers
+		tmp := Rprev
+		Rprev = Rcur
+		Rcur = tmp
+	}
+	return Rprev[k-1], k
+}
+
+//
```

Comment:
Please don't decorate comments.

I'm not sure about this function. We do have similar kind of help elsewhere, but it usually a whole level of abstraction above, not just a slight rename in the same package. I'd rather leave this out and we can add it later if we need (with a better name).

Reply:
I removed the function